gctscript/vm/vm.go

```diff
@@ -0,0 +1,284 @@
+package vm
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/d5/tengo/v2"
+	"github.com/gofrs/uuid"
+	"github.com/thrasher-corp/gocryptotrader/common/crypto"
+	scriptevent "github.com/thrasher-corp/gocryptotrader/database/repository/script"
+	"github.com/thrasher-corp/gocryptotrader/gctscript/modules/loader"
+	"github.com/thrasher-corp/gocryptotrader/gctscript/wrappers/validator"
+	log "github.com/thrasher-corp/gocryptotrader/logger"
+	"github.com/volatiletech/null"
+)
+
+// NewVM attempts to create a new Virtual Machine firstly from pool
+func NewVM() (vm *VM) {
+	newUUID, err := uuid.NewV4()
+	if err != nil {
+		log.Error(log.GCTScriptMgr, Error{
+			Action: "New: UUID",
+			Cause:  err,
+		})
+		return nil
+	}
+
+	if GCTScriptConfig.Verbose {
+		log.Debugln(log.GCTScriptMgr, "New GCTScript VM created")
+	}
+
+	vm = &VM{
+		ID:     newUUID,
+		Script: pool.Get().(*tengo.Script),
+	}
+	vm.event(StatusSuccess, TypeCreate, false)
+	return
+}
+
+// Load parses and creates a new instance of tengo script vm
+func (vm *VM) Load(file string) error {
+	if vm == nil {
```

Comment:
Even though this is fine for production, there's no test coverage for this currently